### PR TITLE
Don't Check for Last Admin on Update

### LINKED ISSUE #4131: Change password for User error - "Removing the last Admin is not allowed"
Your Rocket.Chat version: 0.37.1

If i try to change the password for my test user i get the error "Removing the last Admin is not allowed"
Screenshot:
image
There are no errors in the server log..

### PR DESCRIPTION
**Pull request by graywolf336**
@RocketChat/core 

Closes #4131

There was a check on the `update` part of `insertOrUpdateUser` which was checking, incorrectly, if the user being updated wasn't admin and if they weren't then the update wasn't allowed. Since the `update` part of that method doesn't actually handle roles I removed that check.

### CODE REVIEW

**Comment by sampaiodiego** [comment]
That check is indeed incorrectly, but it need to be fixed instead of removed =)

I fixed the code.. what you think?

**Comment by graywolf336** [comment]
May I ask why it can't be removed? The update method doesn't do anything with regards to the roles

**Comment by engelgabriel** [comment]
We are trying to prevent that the role `admin` is removed from the last admin.

**Comment by graywolf336** [comment]
Right, I understand that but that method does not handle roles on the user after that check so the last admin can't be removed in that method thus that check is not required/needed there...unless I am over looking something in the code

**Comment by sampaiodiego** [comment]
you're right @graywolf336 .. that code [was recently changed](https://github.com/RocketChat/Rocket.Chat/commit/0522d2ca52ba5db130acb2ed36fb7445df44b8c9#diff-ad7a533d660cfd5a606b9ae6b0a0d73aL106) and it does change the role previously.. that's why the confusion.

but now the UI don't even has the "role" field while editing a user... so, agreed, let's remove the test!

**Comment by graywolf336** [comment]
Got it, that makes a lot of sense now. :+1:

**Comment by sampaiodiego** [comment]
LGTM

**Comment by sampaiodiego** [comment]
thanks @timkinnane for your concern..
but [as said](https://github.com/RocketChat/Rocket.Chat/pull/4136#issuecomment-243251646) by @graywolf336 the role isn't updated on this method.. so this check became useless.

**Comment by timkinnane** [comment]
Oh sorry I assumed [this line](https://github.com/RocketChat/Rocket.Chat/blob/c9f00e77340ad3a2b4a84a2e170417981fa3df4f/packages/rocketchat-lib/server/methods/insertOrUpdateUser.coffee#L114) just updated any attributes provided to the method, which could contain role.

It may not update role in current usage, but if there's a chance it could I'd prefer to keep the check because a future change might expose role selection in user edit views, but at that point the idea of the last admin check would probably not be considered. Or is it actually not possible through the `insertOrUpdateUser` method?

**Comment by graywolf336** [comment]
@timkinnane 

That line you link to is passing in an object which is manually built on [line 101](https://github.com/RocketChat/Rocket.Chat/blob/c9f00e77340ad3a2b4a84a2e170417981fa3df4f/packages/rocketchat-lib/server/methods/insertOrUpdateUser.coffee#L101).

While I agree that we need to be cognizant of any future changes which handle roles and verifying the last admin can not be removed, this file at the moment can not *update* a user's role even when that property is passed in since the actual object which updates the user is manually built in the code. Does this make sense?

So in short to answer your last question, it is not possible through the `insertOrUpdateuser` method.

**Comment by timkinnane** [comment]
Yeah thanks for your patience explaining the process. I had missed the distinction between `updateUser` and `userData`.
